addons/stock/models/stock_inventory.py

```diff
@@ -475,17 +476,16 @@ def write(self, vals):
         return res
 
     def _check_no_duplicate_line(self):
+        # Performance: restrict domain on 'not null' fields of stock_inventory_line.
+        domain = [('product_id', 'in', self.product_id.ids), ('location_id', 'in', self.location_id.ids)]
```

Comment:
use child_of, no ?

Reply:
@fmdl But since the key in the `for line in self` line below uses `line.location_id.id`, wouldn't using
child_of in the search domain return inventory_lines that will never raise a UserError ?